cmd/kubeadm/app/apis/kubeadm/v1alpha1/conversion.go

```diff
@@ -46,12 +48,30 @@ func Convert_v1alpha1_MasterConfiguration_To_kubeadm_MasterConfiguration(in *Mas
 
 	UpgradeCloudProvider(in, out)
 	UpgradeAuthorizationModes(in, out)
+	if err := UpgradeBootstrapTokens(in, out); err != nil {
+		return err
+	}
 	// We don't support migrating information from the .PrivilegedPods field which was removed in v1alpha2
 	// We don't support migrating information from the .ImagePullPolicy field which was removed in v1alpha2
 
 	return nil
 }
 
+func Convert_kubeadm_MasterConfiguration_To_v1alpha1_MasterConfiguration(in *kubeadm.MasterConfiguration, out *MasterConfiguration, s conversion.Scope) error {
+	if err := autoConvert_kubeadm_MasterConfiguration_To_v1alpha1_MasterConfiguration(in, out, s); err != nil {
+		return err
+	}
+
+	if len(in.BootstrapTokens) > 0 {
+		out.Token = in.BootstrapTokens[0].Token.String()
```

Comment:
I think it should be useful to add a comment to take note that only the first token is preserved in this conversion (and eventually glog the discarded tokens)

Reply:
I'll add better comments that that conversion is never supported. We don't support downgrading the API for that exact reason. It's only there for the fuzzing tests.

Reply:
done